### PR TITLE
[Feature] Support instance  list command

### PR DESCRIPTION
**Pull request by arugal**
works like this:
```bash
./swctl --debug --base-url=http://122.112.182.72:8080/graphql --display=yaml instance ls --service-id=4 --start=2019-11-09 --step=MONTH
DEBUG Debug mode is enabled                        
DEBUG Using configurations:
 global:
   base-url: http://122.112.182.72:8080/graphql 
DEBUG Start time: 2019-11-09 end time:             
DEBUG >> variables: map[duration:{2019-11-09 2019-12-09 DAY} serviceId:4] 
DEBUG >> query: 
                query ($serviceId: ID!, $duration: Duration!) {
                instances: getServiceInstances(duration: $duration, serviceId: $serviceId) {
                        id
                        name
                        language
                                instanceUUID
                        attributes {
                                name
                                value
                        }
                        }
                }
         
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]] 
DEBUG << {"data":{"instances":[{"id":"4","name":"projectA-pid:4771@skywalking-server-0001","language":"JAVA","instanceUUID":"d54c4f5b5094419c9f0d12b8445730c0","attributes":[{"name":"os_name","value":"Linux"},{"name":"host_name","value":"skywalking-server-0001"},{"name":"process_no","value":"4771"},{"name":"ipv4s","value":"192.168.252.12"}]}]}} 
- id: "4"
  name: projectA-pid:4771@skywalking-server-0001
  attributes:
  - name: os_name
    value: Linux
  - name: host_name
    value: skywalking-server-0001
  - name: process_no
    value: "4771"
  - name: ipv4s
    value: 192.168.252.12
  language: JAVA
  instanceuuid: d54c4f5b5094419c9f0d12b8445730c0

================================================

./swctl --debug --base-url=http://122.112.182.72:8080/graphql --display=yaml instance ls --service-name=projectB --start=2019-11-09 --step=MONTH

DEBUG Debug mode is enabled                        
DEBUG Using configurations:
 global:
   base-url: http://122.112.182.72:8080/graphql 
DEBUG Start time: 2019-11-09 end time:             
DEBUG >> variables: map[serviceCode:projectB]      
DEBUG >> query: 
                query searchService($serviceCode: String!) {
                service: searchService(serviceCode: $serviceCode) {
                                id name
                        }
                }
         
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]] 
DEBUG << {"data":{"service":{"id":"2","name":"projectB"}}} 
DEBUG >> variables: map[duration:{2019-11-09 2019-12-09 DAY} serviceId:2] 
DEBUG >> query: 
                query ($serviceId: ID!, $duration: Duration!) {
                instances: getServiceInstances(duration: $duration, serviceId: $serviceId) {
                        id
                        name
                        language
                                instanceUUID
                        attributes {
                                name
                                value
                        }
                        }
                }
         
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]] 
DEBUG << {"data":{"instances":[{"id":"2","name":"projectB-pid:4713@skywalking-server-0001","language":"JAVA","instanceUUID":"55a7a0af57f64c1da0b7c44063ddc8c0","attributes":[{"name":"os_name","value":"Linux"},{"name":"host_name","value":"skywalking-server-0001"},{"name":"process_no","value":"4713"},{"name":"ipv4s","value":"192.168.252.12"}]}]}} 
- id: "2"
  name: projectB-pid:4713@skywalking-server-0001
  attributes:
  - name: os_name
    value: Linux
  - name: host_name
    value: skywalking-server-0001
  - name: process_no
    value: "4713"
  - name: ipv4s
    value: 192.168.252.12
  language: JAVA
  instanceuuid: 55a7a0af57f64c1da0b7c44063ddc8c0
================================================================
./swctl --debug --base-url=http://122.112.182.72:8080/graphql --display=yaml instance ls --service-name=projectB --service-id=4 --start=2019-11-09 --step=MONTH

DEBUG Debug mode is enabled                        
DEBUG Using configurations:
 global:
   base-url: http://122.112.182.72:8080/graphql 
DEBUG Start time: 2019-11-09 end time:             
DEBUG >> variables: map[duration:{2019-11-09 2019-12-09 DAY} serviceId:4] 
DEBUG >> query: 
                query ($serviceId: ID!, $duration: Duration!) {
                instances: getServiceInstances(duration: $duration, serviceId: $serviceId) {
                        id
                        name
                        language
                                instanceUUID
                        attributes {
                                name
                                value
                        }
                        }
                }
         
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]] 
DEBUG << {"data":{"instances":[{"id":"4","name":"projectA-pid:4771@skywalking-server-0001","language":"JAVA","instanceUUID":"d54c4f5b5094419c9f0d12b8445730c0","attributes":[{"name":"os_name","value":"Linux"},{"name":"host_name","value":"skywalking-server-0001"},{"name":"process_no","value":"4771"},{"name":"ipv4s","value":"192.168.252.12"}]}]}} 
- id: "4"
  name: projectA-pid:4771@skywalking-server-0001
  attributes:
  - name: os_name
    value: Linux
  - name: host_name
    value: skywalking-server-0001
  - name: process_no
    value: "4771"
  - name: ipv4s
    value: 192.168.252.12
  language: JAVA
  instanceuuid: d54c4f5b5094419c9f0d12b8445730c0
```
*Set both ```--service-id``` and ```--service-name```,  ```--service-name``` is ignored*

### CODE REVIEW

**Comment by arugal** [comment]
help works like this:
```bash
./bin/swctl i ls -h 

NAME:
   swctl instance list - List all available instance by given --service parameter

USAGE:
   swctl instance list [command options] [arguments...]

OPTIONS:
   --start TIME  query start TIME
   --end TIME    query end TIME
   --service ID  query service ID

```

File: graphql/client/client.go [hunk 1]
```diff
@@ -50,3 +55,33 @@ func Services(cliCtx *cli.Context, duration schema.Duration) []schema.Service {
...
+			}
+		}
+	`)
+	request.Var("serviceId", serviceId)
+	request.Var("duration", duration)
+
+	ctx := context.Background()
+	if err := client.Run(ctx, request, &response); err != nil {
+		logger.Log.Fatalln(err)
+		panic(err)
```
Comment (kezhenxu94) [inline graphql/client/client.go]: Seems we don't need `panic`, `logger.Log.Fatalln(err)` calls `os.exit`, and `panic` is unreachable

Comment (arugal) [reply graphql/client/client.go]: I'll check all the code

File: swctl/main.go [hunk 2]
```diff
@@ -60,12 +61,14 @@ func main() {
 		altsrc.NewStringFlag(cli.StringFlag{
 			Name:     "display",
 			Required: false,
-			Usage:    "display `style` of the result, supported styles are: json, yaml",
+			Usage:    "display `style` of the result, supported styles are: json, yaml, table",
+			Value:    "yaml",
```
Comment (kezhenxu94) [inline swctl/main.go]: Good catch, but I think the default should be JSON, it's more useful in to pipe commands

**Comment by wu-sheng** [comment]
As I commented at the issue, how could do separated ID and name in --service both? This seems not right.

**Comment by arugal** [comment]
@wu-sheng  Support both Name and ID in the latest code.

File: swctl/main.go (hunk 2 above)
Comment (arugal) [reply swctl/main.go]: done

File: graphql/client/client.go (hunk 1 above)
Comment (arugal) [reply graphql/client/client.go]: done

**Comment by wu-sheng** [comment]
How do you know it is ID? Try to parse int?

**Comment by arugal** [comment]
> How do you know it is ID? Try to parse int?

First through the name search services. If the query result is not empty, get serviceId from the query result, or else take the ```--service value``` as the serviceId

**Comment by wu-sheng** [comment]
> First through the --service value search services. If the query result is not empty, get serviceId from the query result, or else take the --service value as the serviceId

If a service name is an integer, then you may face ambiguity issue. 
Such as `swctl instance --service 123` meaning `service name = 123` or service id = `123`?

I would say `swctl instance --service 123` must be `service name = 123`. If id is expected to be supported, choose maybe `swctl instance --service-id 123` or something we could discuss later. 

CLI is a command line tool, I want it has clear meaning.

**Comment by arugal** [comment]
> > First through the --service value search services. If the query result is not empty, get serviceId from the query result, or else take the --service value as the serviceId
> 
> If a service name is an integer, then you may face ambiguity issue.
> Such as `swctl instance --service 123` meaning `service name = 123` or service id = `123`?
> 
> I would say `swctl instance --service 123` must be `service name = 123`. If id is expected to be supported, choose maybe `swctl instance --service-id 123` or something we could discuss later.
> 
> CLI is a command line tool, I want it has clear meaning.

Agree to provide ```--service-name``` and ```--service-id```.

File: graphql/client/client.go [hunk 3]
```diff
@@ -42,11 +54,57 @@ func Services(cliCtx *cli.Context, duration schema.Duration) []schema.Service {
...
+
+	executeQuery(cliCtx, request, &response)
+	return response["instances"]
+}
+
+func searchServices(cliCtx *cli.Context, serviceName string, duration schema.Duration) (service schema.Service, err error) {
+	var response map[string][]schema.Service
+	request := graphql.NewRequest(`
+		query searchServices($keyword: String!, $duration: Duration!) {
+    		service: searchServices(duration: $duration, keyword: $keyword) {
```
Comment (wu-sheng) [inline graphql/client/client.go]: You should not use searchService. Use `searchService` by using service name as the parameter, this should be an equal match.

Comment (arugal) [reply graphql/client/client.go]: done

**Comment by arugal** [comment]
@wu-sheng @kezhenxu94  please review

**Comment by arugal** [comment]
I will submit a new PR to resolve conflicts.

**Comment by wu-sheng** [comment]
Why do you need a new PR? You could update this branch and PR. New PR will make the context lose.

**Comment by wu-sheng** [comment]
We are using squash and merge, you should not worry about multiple commit logs.

**Comment by wu-sheng** [comment]
@kezhenxu94 @heyanlong Please recheck.

**Pull request merged**